src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.Async.cs

```diff
@@ -0,0 +1,254 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression;
+
+internal sealed partial class ZipGenericExtraField
+{
+    public async Task WriteBlockAsync(Stream stream, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        byte[] extraFieldHeader = new byte[SizeOfHeader];
```

Comment:
We could use ArrayPools here (and in other places) if we want, but we can easily switch to using them in a later PR if you want.

Reply:
SizeIfHeader is small enough that allocating a regular byte might be sufficient.